app/models/newsletter_finder.rb

```diff
@@ -0,0 +1,27 @@
+class NewsletterFinder
```

Comment:
It looks like there is the concept of `newsletters` and `flyers`. Have you thought of separating these out? It would not only solve your "open/closed principle" violation as @halogenandtoast suggested, but also make it clear what is being duplicated that you could possibly extract into a new parent class.